certora/specs/StakePool.spec

```diff
@@ -325,25 +326,34 @@ rule withdrawalAtLeastMinToken(env e){
     assert amount < minWithdrawal => lastReverted;
 }
 
- rule initAllowedOnlyOnce(method f){
+ // Should be initialize once only.
+rule initAllowedOnlyOnce(method f){
     env e; env e1;
     calldataarg args; calldataarg args1;
     initialize(e,args);
     initialize@withrevert(e1,args1);
     bool reverted = lastReverted;
     assert reverted;
- }
-
-
-/*************** NEED MORE WORK *******/
+}
 
-/*** Rules that might not be accurate - they have violations ***/
-/*
-rule bnbToUnbondAndBnbUnboundingCorrelation(method f, address user) filtered {f-> f.selector != initialize(address,(address,uint256,uint256,uint256,(uint256,uint256,uint256))).selector }
+// The change in bnbToUnbond should impact on bnbUnbonding
+rule bnbToUnbondAndBnbUnboundingCorrelation(method f, address user)filtered {f-> f.selector != initialize(address,(address,uint256,uint256,uint256,uint256,uint256,(uint256,uint256,uint256))).selector }
 {
     env e;
     require user == e.msg.sender && user != currentContract;
+ 
+    uint256 rewardFee;
+    uint256 depositFee;
+    uint256 withdrawFee;
     
+    rewardFee, depositFee, withdrawFee = getFee();
+    // assumptions 
+    require rewardFee == 0 && depositFee < MAX_FEE() && withdrawFee == 0;
```

Comment:
Here too can use the same function, no need to restrict reward and withdraw fee to 0

Reply:
Done